### PR TITLE
Move user related model into models/user

### PR DESCRIPTION
**Pull request by lunny**
As title.

### CODE REVIEW

**Comment by wxiaoguang** [comment]
Is `models/db/name.go` in the right package?

**Comment by lunny** [comment]
> Is `models/db/name.go` in the right package?

The variables and functions in that files will be shared for other models/user and future models/repo packages. Or we have to create a new sub package to do that.

**Comment by wxiaoguang** [comment]
OK, then we can have another PR to continue the refactoring, this PR is already so large. 

LGTM

**Review by wxiaoguang — APPROVED** [review]

File: models/lfs_lock.go [hunk 1]
```diff
@@ -22,7 +23,6 @@ type LFSLock struct {
 	ID      int64       `xorm:"pk autoincr"`
 	Repo    *Repository `xorm:"-"`
 	RepoID  int64       `xorm:"INDEX NOT NULL"`
-	Owner   *User       `xorm:"-"`
```
Comment (KN4CK3R) [inline models/lfs_lock.go:25]: Is the linter able to detect unused members?

Comment (Gusted) [reply models/lfs_lock.go:25]: The linter is able to detect unused private members, but exported one like these aren't detected.

I think the linter is preventing to detect this, as this Field could be used when go-gitea/gitea is imported as package and a golang program is using that field. But given that go-gitea/gitea shouldn't be used as package(right?), we can modify the deadcode linter and be more aggressive and also check on these Public fields etc. Not sure which refactor PR it was mentioned, but we also could check if a certain function is only used with the `xxx_test.go` packages and move it to them.

https://github.com/golangci/go-misc/blob/master/deadcode/deadcode.go

**Review by KN4CK3R — APPROVED** [review]

File: models/lfs_lock.go (hunk 1 above)
Comment (KN4CK3R) [reply models/lfs_lock.go:25]: Currently we expect no one imports gitea. Otherwise this change would not be possible as we break the "api".

Comment (Gusted) [reply models/lfs_lock.go:25]: Got it, I will open up a issue for such proposal. Should I open up a issue at [gitea-vet](https://gitea.com/gitea/gitea-vet/issues) or just here.

**Comment by lunny** [comment]
make L-G-T-M work

**Pull request merged**